### PR TITLE
[minor] Add support for s390x architecture

### PR DESCRIPTION
**Pull request by shajeena**
Changes are made to support multiarchitecture support (s390x) in cli-base

Can you please review the changes

### CODE REVIEW

File: build/bin/.functions.sh [hunk 1]
```diff
@@ -50,7 +50,40 @@ function install_yq() {
...
+    sudo apt-get install -y qemu-user-static
+      echo "inside2 install_buildx"
+    qemu-aarch64-static --version
+echo "inside3 install_buildx"
+    sudo apt-get install -y binfmt-support
+    update-binfmts --version
+
+    docker buildx create --name mybuilder
+    docker buildx use mybuilder
+#  fi
```
Comment (whitfiea) [inline build/bin/.functions.sh]: Indenting looks wrong here and we have a commented out if block, that should be deleted if not used

File: build/bin/docker-build.sh [hunk 2]
```diff
@@ -26,6 +26,10 @@ do
         DOCKERFILE="$2"
         ;;
 
+    --target-platform)
+    TARGET_PLATFORM="$2"
+    ;;
+
```
Comment (whitfiea) [inline build/bin/docker-build.sh:32]: Indent is wrong

File: build/bin/docker-build.sh [hunk 3]
```diff
@@ -48,10 +57,32 @@ echo "VERSION_LABEL .. $DOCKER_TAG"
...
+  else LOCAL_TAG=$NAMESPACE/$IMAGE:$DOCKER_TAG
+  fi
 docker build \
+  --build-arg ARCHITECTURE=amd64 \
   --build-arg VERSION_LABEL=$DOCKER_TAG \
   --build-arg RELEASE_LABEL=$GITHUB_RUN_ID \
   --build-arg VCS_REF=$GITHUB_SHA \
   --build-arg VCS_URL=https://github.com/$GITHUB_REPOSITORY \
-  -t $NAMESPACE/$IMAGE $EXTRA_PARAMS -f $DOCKERFILE $BUILDPATH
+   -t $LOCAL_TAG $EXTRA_PARAMS -f $DOCKERFILE $BUILDPATH
```
Comment (whitfiea) [inline build/bin/docker-build.sh]: Needs to be indented now it is in an if block

File: image/cli-base/Dockerfile [hunk 4]
```diff
@@ -13,29 +14,37 @@ RUN dnf update -y --skip-broken --nobest &&\
 
 # 2. Upgrade pip, install wheel, then install Python modules
 USER default
+
 COPY --chown=default:root install /tmp/install
 RUN python3 -m pip install --no-cache-dir --upgrade pip setuptools wheel &&\
-    python3 -m pip install --no-cache-dir -r /tmp/install/requirements.txt
+    pip3 install -r /tmp/install/requirements.txt --find-links /tmp/install
+    #python3 -m pip install --no-cache-dir -r /tmp/install/requirements.txt
```
Comment (whitfiea) [inline image/cli-base/Dockerfile:21]: delete commented out code

File: image/cli-base/Dockerfile [hunk 5]
```diff
@@ -13,29 +14,37 @@ RUN dnf update -y --skip-broken --nobest &&\
...
+
+    bash /tmp/install/install-helm.sh  --target-platform $ARCHITECTURE && \
+    bash /tmp/install/install-mongo-tools.sh  --target-platform $ARCHITECTURE && \
+    bash /tmp/install/install-yq.sh  --target-platform $ARCHITECTURE && \
+    bash /tmp/install/install-argocd.sh --target-platform $ARCHITECTURE
+
+RUN if [ "${ARCHITECTURE}" == "amd64" ] ; then \
+         bash /tmp/install/install-aws.sh; \
+         bash /tmp/install/install-rosa.sh; \
+         bash /tmp/install/install-rclone.sh; \
```
Comment (whitfiea) [inline image/cli-base/Dockerfile:40]: rclone is needed for the backup/restore. Is that not going to be available in s390x?

File: image/cli-base/install/install-mongo-tools.sh [hunk 6]
```diff
@@ -1,20 +1,52 @@
...
+  tar xvfz mongodb-database-tools-rhel*.tgz
+  mv mongodb-database-tools-rhel*/bin/* /usr/local/bin/
+  rm -rf mongodb-database-tools-rhel*
 mongodump --version
+#mongodump version: 100.9.5
+ #git version: 90481484c1783826fe26ca18bbdcd30e933f3b88
+ #Go version: go1.21.11
+ #   os: linux
+ #   arch: s390x
+ #   compiler: gc
```
Comment (whitfiea) [inline image/cli-base/install/install-mongo-tools.sh:52]: remove commented out code

File: image/cli-base/install/install-rclone.sh [hunk 7]
```diff
@@ -11,3 +11,7 @@ cp ./rclone-*-linux-amd64/rclone /usr/local/bin/
 rclone version
 
 rm -rf rclone-*
+
+#s390x not supported
+#[root@aca72e69ab39 rclone-v1.65.2-linux-mipsle]# rclone version
+ #bash: /usr/local/bin/rclone: cannot execute binary file: Exec format error
```
Comment (whitfiea) [inline image/cli-base/install/install-rclone.sh:17]: How will the backup/restore function work?

**Review by whitfiea — CHANGES_REQUESTED** [review]
Can the PR title be updated to be more meaningful.

Also the PR should show that you have run with this cli-base in a cli version that works on s390x

File: image/cli-base/install/install-rclone.sh (hunk 7 above)
Comment (durera) [reply image/cli-base/install/install-rclone.sh:17]: Initially we are going to state limited support for running on s390x ... the README in this repo needs to be updated with the details of what restrictions are in place in the s390x image ... e.g. we're expecting that neither aws or rclone will be included on the s390x base image; and when we build the s390x cli image we will update it's README to document the features of the cli that are not supported on s390x.

File: image/cli-base/Dockerfile (hunk 5 above)
Comment (durera) [reply image/cli-base/Dockerfile:40]: Initially, yes.  

> Initially we are going to state limited support for running on s390x ... the README in this repo needs to be updated with the details of what restrictions are in place in the s390x image ... e.g. we're expecting that neither aws or rclone will be included on the s390x base image; and when we build the s390x cli image we will update it's README to document the features of the cli that are not supported on s390x.

**Review by durera — CHANGES_REQUESTED** [review]
In addition to @whitfiea's review comments, reminder that the README must be updated with details of the restrictions in the s390x image so that it is clear what is included in amd64 versus s390x.

I suggest converting the list into a table, something like:

| content | amd64 | s390x |
| --- | --- | --- |
| aws | ✔️ | ❌ |
| rclone | ✔️ | ❌ |
| whatever | ✔️ | ✔️ |

**Comment by durera** [comment]
Closing PR.  Superseded by https://github.com/ibm-mas/cli-base/pull/13

**Pull request closed without merging**